### PR TITLE
feat: HIP-584 Historical: Sync evm configurations with hedera-services

### LINKED ISSUE #7384: HIP-584: Historical Support - Sync EVM configurations with hedera-services ContractsModule
### Problem

After testing historical support loading of evm version 30 - historical calls are failing since for 
version 30 we are not populating evm precompiled contracts.

Also our default EVM object config is version 38 and we are missing the config for version 34.

We need to sync these evm configs with hedera-services ContractsModule.

### Solution

Currently we only use one EVM bean that does **registerShanghaiOperations** -> v38 version

Add 3 versions of EVM bean

v30 - **registerLondonOperations**

v34 - **registerParisOperations**

v38 - **registerShanghaiOperations**

Configure **MessageCallProcessor** to register precompiledContracts for version 30 and use v30 EVM bean.

Add new **ContractCreationProcessor** bean that will use new v30 EVM

Add new **HederaEvmMessageCallProcessor** bean that will use new v34 EVM bean.

Add new **ContractCreationProcessor** bean that will use new v34 EVM.

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by kselveliev**
**Description**:
This PR modifies current EVM bean configuration - which defaults to evm 38 

Create 3 versions of **EVM** bean - similar to services
v30 - registerLondonOperations
v34 - registerParisOperations
v38 - registerShanghaiOperations

Changed config for **MessageCallProcessor** that is used for **EVM_VERSION_30** since it was missing to register precompiled contracts -> created **MirrorEvmMessageCallProcessorV30** - which now accepts v30 EVM bean and registers the precompiles provided from **PrecompiledContractProvider**.
Added **ContractCreationProcessorV30** that uses new v30 EVM bean

Added **MirrorEvmMessageCallProcessorV34** that uses v34 EVM bean.
Added **ContractCreationProcessorV34** that uses v34 EVM bean.

Added **MirrorEvmMessageCallProcessorV38** that uses v38 EVM bean.
Added **ContractCreationProcessorV38** that uses v38 EVM bean. (NB: This was the only evm config that was loaded before the changes in the PR)

Modified **EvmConfiguration** registering the 3 versions of evm v30, v34 and v38.
And modified the messageCallProcessors and contractCreationProcessorProvider to map the evm version to
the newly created configs.

image

image

**Related issue(s)**:

Fixes #7384

**Notes for reviewer**:

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/config/EvmConfiguration.java [hunk 1]
```diff
@@ -226,17 +238,82 @@ gasCalculator, mirrorNodeEvmProperties, createOperationExternalizer()),
...
-        return new HederaPrngSeedOperation(gasCalculator, prngLogic);
+    EVM v34(
+            final GasCalculatorHederaV22 gasCalculator,
+            final MirrorNodeEvmProperties mirrorNodeEvmProperties,
+            final HederaPrngSeedOperation prngSeedOperation,
+            final HederaBlockHashOperation hederaBlockHashOperation) {
+        final var operationRegistry = new OperationRegistry();
+        final BiPredicate<Address, MessageFrame> validator = (Address x, MessageFrame y) -> true;
+
+        registerParisOperations(
```
Comment (steven-sheehy) [inline hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/config/EvmConfiguration.java:253]: The 3 EVMs duplicate their configuration except for this register method and the EvmSpecVersion. Create a functional interface like 

```java
interface OperationRegistryCallback {
    void register(OperationRegistry registry, GasCalculator gasCalculator, BigInteger chainId);
}
```

Then declare most of the parameters as class variables and create a helper method:

```java
    private EVM evm(HederaPrngSeedOperation prngSeedOperation, EvmSpecVersion evmSpecVersion, OperationRegistryCallback callback) {
...
    }

    @Bean
    EVM evm34(HederaPrngSeedOperation prngSeedOperation) {
        return evm(prngSeedOperation, EvmSpecVersion.PARIS, MainnetEVMs::registerParisOperations);
    }
```

File: hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/contracts/execution/MirrorEvmMessageCallProcessorV34.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+import org.hyperledger.besu.evm.EVM;
+import org.hyperledger.besu.evm.precompile.PrecompileContractRegistry;
+
+@Named
+public class MirrorEvmMessageCallProcessorV34 extends MirrorEvmMessageCallProcessor {
+
+    public MirrorEvmMessageCallProcessorV34(
+            AbstractAutoCreationLogic autoCreationLogic,
+            EntityAddressSequencer entityAddressSequencer,
+            @Named("v34") EVM v34,
```
Comment (steven-sheehy) [inline hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/contracts/execution/MirrorEvmMessageCallProcessorV34.java:33]: MirrorEvmMessageCallProcessor 34 and 38 are identical except the EVM so can be moved to config for code reuse like ContractCreationProcessor.

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

File: hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/config/EvmConfiguration.java [hunk 3]
```diff
@@ -195,7 +207,7 @@ public boolean shouldFailBasedOnLazyCreation(final MessageFrame frame, final Add
     }
 
     @Bean
-    EVM evm(
+    EVM v38(
```
[the remark this answers is not in the document]
Comment (kselveliev) [reply hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/config/EvmConfiguration.java]: Comment addressed. Now using names evm030, evm034, evm038

File: hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/contracts/execution/ContractCreationProcessorV30.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+import java.util.List;
+import org.hyperledger.besu.evm.EVM;
+import org.hyperledger.besu.evm.gascalculator.GasCalculator;
+import org.hyperledger.besu.evm.processor.ContractCreationProcessor;
+
+@Named
+public class ContractCreationProcessorV30 extends ContractCreationProcessor {
+
+    public ContractCreationProcessorV30(GasCalculator gasCalculator, @Named("v30") EVM v30) {
+        super(gasCalculator, v30, true, List.of(), 1);
```
[the remark this answers is not in the document]
Comment (kselveliev) [reply hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/contracts/execution/ContractCreationProcessorV30.java]: Comment addressed. Now using Qualifier for evm bean in constructor.

File: hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/config/EvmConfiguration.java (hunk 1 above)
Comment (kselveliev) [reply hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/config/EvmConfiguration.java]: Comment addressed

File: hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/contracts/execution/MirrorEvmMessageCallProcessorV34.java (hunk 2 above)
Comment (kselveliev) [reply hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/contracts/execution/MirrorEvmMessageCallProcessorV34.java]: Comment addressed. Applied the change similar to ContractCreationProcessor above

**Review by steven-sheehy — APPROVED** [review]
LGTM

**Pull request merged**